coderd/templates.go

```diff
@@ -855,6 +861,35 @@ func (api *API) patchTemplateMeta(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, api.convertTemplate(updated))
 }
 
+func (api *API) notifyUsersOfTemplateDeprecation(ctx context.Context, template database.Template) error {
+	users, err := api.Database.GetUsersWithAccessToTemplateByID(ctx, template.ID)
```

Comment:
This is technically as specified in the issue description, but I feel like this over-delivers. If there are 1k users in an organization but only a few actually use it, it doesn't seem useful to send email notifications to the other users.

How about we instead query for workspaces that reference the template and notify the (deduplicated) owners of those workspaces?

Pros: 
- Fewer notifications sent in the worst-case scenario
- This shouldn't require a separate query

Cons:
- Not **specifically** what was asked for
- `GetWorkspaces` is paginated which may be slightly annoying

Reply:
Yeah I agree that only notifying users of the template is the better choice.

Reply:
> `GetWorkspaces` is paginated which may be slightly annoying

Reading the query it looks like it only paginates if `GetWorkspacesParams.Limit != 0`, otherwise it returns everything.

https://github.com/coder/coder/blob/bcd68ee24953d4148e12cd3ab6c05d49db090c36/coderd/database/queries/workspaces.sql#L340-L344

Reply:
That actually looks like a syntax error if limit == 0. You can't use the keyword LIMIT without a value.

Reply:
I thought the same but it turns out this is fine! The `CASE ... END` will return a `NULL` on the case where `limit == 0`. `LIMIT NULL` is equivalent to no call to `LIMIT`.

https://www.postgresql.org/docs/current/queries-limit.html
https://www.postgresql.org/docs/current/functions-conditional.html#FUNCTIONS-CASE